django/db/backends/mysql/schema.py

```diff
@@ -62,6 +62,23 @@ def _model_indexes_sql(self, model):
                     field.db_index = False
         return super(DatabaseSchemaEditor, self)._model_indexes_sql(model)
 
+    def _delete_unique_together(self, model, fields):
+        """
+        Believe it or not, MySQL can remove an implicit FK index on a field when
+        that field is covered by another index like a unique_together.
+        http://bugs.mysql.com/bug.php?id=37910 / Django ticket #24757
+        We check here before removing the unique_together if we have to recreate
+        an FK index.
+        """
+        first_field = model._meta.get_field(fields[0])
```

Comment:
I was a bit surprised that the ordering seemed to matter (e.g. there's no problem if the FK isn't first in unique_together). Might be helpful to add a comment about that if you know the reason.